#159 Release Notes 1.5.0 (#172)
Inbibo/adonisfx_docs: docs/release_notes.md

# Release Notes

## Version 1.5.0

### Core

- Implemented a relaxation algorithm to smooth creases, over compression and over stretching from a geometry surface.

### Maya

> - Added support to Maya 2025.

Reviewer: I think this is more correct. Sorry for that ;)
```suggestion
- Added support for Maya 2025.
```

- Implemented an experimental Python API.

> - Implemented a mirroring script in Python to transfer the muscle setup (muscles and sensors) from one side of the character to the other.

Reviewer 2: Questions. the locators would be also mirrored?

- Added AdnRelax deformer that applies the relaxation algorithm to the given geometry.
- Added fibers multiplier map to AdnMuscle and AdnRibbonMuscle to control the distribution of the activation across the surface of a muscle.
- Added a utility to create AdnActivation nodes from the menu.
- Added a utility to remove inputs from an AdnActivation node.
- Prevent painting fibers and tendons on a simulated frame.

### Improvements

- Added further optimizations to AdnSimshape solver.
- Added further optimizations to AdnSkin solver.

> - Improved the automation of the Paint Tool to include the AdnDebugLocator in the current selection to easy the debugging process while painting.

Reviewer 2: Maybe this is a bit more correct?
```suggestion
- Improved the automation of the Paint Tool to include the AdnDebugLocator in the current selection to ease the debugging process while painting.
```

Author: Correct, fixed

### Bug Fixes

- Fixed a bug that prevented to generate the fibers flow driven by the tendon weights when a Maya preset was applied onto a muscle deformer. *AdonisFX-1652*

### Deprecated

- The Exporter and Importer tools are deprecated and removed from the UI. The internal logic in Python is still available but it will be removed or rewritten in an upcoming release.

> ### Known Limitations

Reviewer 2: Should we are here or somewhere else something about the Cached Playback?

Author: You are totally right. I have added two limitatnions here:

- Maya Cached Playback can be used for simulation preview but in order to get the correct simulation results it is required to disable the Cached Playback. *AdonisFX-1624*
- Features debugging in the viewport is not supported with Maya Cached Playback enabled. *AdonisFX-1600*

> - Painting functionalities on AdnGlue are not supported if other deformers are applied to the AdnGlue output mesh directly. *AdonisFX-1644*

Reviewer 2: Could we say "weight maps" instead of "functionalities"?

Author: Yes, I agree

# Release Notes

## Version 1.5.0

### Core

- Implemented a relaxation algorithm to smooth creases, over compression and over stretching from a geometry surface.

### Maya

- Added support to Maya 2025.
- Implemented an experimental Python API.

> - Implemented a mirroring script in Python to transfer the setup (muscles, locators and sensors) from one side of the character to the other.

Reviewer: Reminder to add the limitation that it's currently only supported for AdnMuscle

Author: I will include this limitation in the page where we will explain how to use the mirroring. Thanks for the reminder

- Added AdnRelax deformer that applies the relaxation algorithm to the given geometry.

> - Added fibers multiplier map to AdnMuscle and AdnRibbonMuscle to control the distribution of the activation across the surface of a muscle.

Reviewer: I would remove "surface" here as we are also taking care of volumetric properties for AdnMuscle
```suggestion
- Added fibers multiplier map to AdnMuscle and AdnRibbonMuscle to control the distribution of the activation across the muscle.
```

- Added a utility to create AdnActivation nodes from the menu.
- Added a utility to remove inputs from an AdnActivation node.

> - Prevent painting fibers and tendons on a simulated frame.

Reviewer: We know that it's not exactly on a simulated frame but I would not know how to write this better. Here a suggestion though.
```suggestion
- Prevent the Paint Tool UI from displaying fibers and tendons attributes on a a simulated frame.
```

Author: What about:

`Prevent the Paint Tool UI from selecting fibers and tendons attributes on a simulated frame.`

### Improvements

- Added further optimizations to AdnSimshape solver.
- Added further optimizations to AdnSkin solver.
- Improved the automation of the Paint Tool to include the AdnDebugLocator in the current selection to ease the debugging process while painting.

### Bug Fixes

> - Fixed a bug that prevented to generate the fibers flow driven by the tendon weights when a Maya preset was applied onto a muscle deformer. *AdonisFX-1652*

Reviewer: Should we reformulate this? The issue was more related on the fibers flow recomputation being triggered on scene open whenever users had applied presets and having had saved the scene out with combed fibers. From reading the sentence above it is not clear to me that this was actually the fix

Author: What about this:
- Fixed a bug that was triggering the recomputation of the fibers flow on scene open when a Maya preset was applied onto the muscle deformer where the source muscle which the user saved the preset from had a different vertex count.

Reviewer: We have already agreed on this!

### Deprecated

> - The Exporter and Importer tools are deprecated and removed from the UI. The internal logic in Python is still available but it will be removed or rewritten in an upcoming release.

Reviewer: I would add a note here saying that it is not advisible to use the code even if the internal logic is still there.

Author: Added: The Exporter and Importer tools are deprecated and removed from the UI. The internal logic in Python is still available but it will be removed or rewritten in an upcoming release. Continuing using it is not advisable.
